src/datachain/lib/dc.py

```diff
@@ -1224,14 +1224,11 @@ def to_pandas(self, flatten=False) -> "pd.DataFrame":
         """
         headers, max_length = self._effective_signals_schema.get_headers_with_length()
         if flatten or max_length < 2:
-            columns = []
-            if headers:
-                columns = [".".join(filter(None, header)) for header in headers]
-            return pd.DataFrame.from_records(self.to_records(), columns=columns)
```

Comment:
So the problem was this line (passing data as a dict and then assigning different column names)? Is the rest refactoring? Just want to make sure I understand the bug and the fix.